### PR TITLE
Issue #17 and #34 initial steps

### PR DESCRIPTION
**Pull request by klauffer**
- Moving the First of many Repository methods into async versions.  

- I didn't want to wrap the synchronous calls in an asynchronous method nor visa-versa as to avoid blocking issues. I chose the least of the evils and went the route of maintaining 2 separate implementations of a method;  synchronous and asynchronous.

- I chose to utilize partial classes to maintain a separation between the synchronous and the asynchronous implementations. this will provide three benefits. 
1. We keep the file sizes from doubling. These files are already chonky.
2. Its Clearer as we are working and refactoring more of these asynchronous methods as to what is or isn't done yet.
3. This is the most important one. when we are done creating all of the async methods we are able to simply delete the deprecated synchronous file. 

- One more Item to bring up. lets say I have a signature like
`Task<IResult<List<T>>> BulkCreateAsync(...) `
but the current Do/Try pattern  makes the implementation look like
`Task<IResult<List<T>>> BulkCreateAsync(...) => Do<Task<List<T>>>.Try((r) =>`
Notice how the return type of the Try method is `Task<List<T>>`. the Try method then wraps that return type in an `IResult`. making the final return type `IResult<Task<List<T>>>` instead of the more appropriate `Task<IResult<List<T>>>`. so the above implementation code errors.
The problem is the Do/Try Pattern encapsulates the return type where in the case of a async Task the return type must be encapsulated. This problem was becoming scope creep for this task and I avoided it as you can see I chose to not utilize the Do/Try wrapper.

### CODE REVIEW

**Review by jstevenson — APPROVED** [review]
The changes look great. I think you have a good solution here and though I haven't used this project in a while, the changes make sense.

**Comment by sysready** [comment]
With respect to the issue surrounding the current Do/Try pattern, my thought would be to implement an async version of the Try method which could be awaited.  While I wasn't able to conduct an actual test, I think this would allow the return type to be correctly wrapped in a task.  An example would be:

`
public static async Task<Do<T>> Try(ILogger logger, Func<IResult<T>, T> workload)
{
	var d = new Do<T>(workload);

	try
	{
		d.Result.ResultObject = workload(d.Result);
	}
	catch (Exception ex)
	{
		d.Exception = ex;
		d.Result.AddExceptionError(ex.GetType().Name, ex);
		logger?.LogError(d.Result.ListErrors());
	}

	return await Task.Run(() => d);
}

public static async Task<Do<T>> Try(Func<IResult<T>, T> workload) => await Try(logger: null, workload: workload);
`

File: test/Conductors.Tests/RepositoryConductorTests/IRepositoryMock.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+using System.Threading.Tasks;
+using AndcultureCode.CSharp.Core.Interfaces;
+using AndcultureCode.CSharp.Core.Interfaces.Conductors;
+using AndcultureCode.CSharp.Core.Interfaces.Data;
+using AndcultureCode.CSharp.Core.Interfaces.Entity;
+using Moq;
+
+namespace AndcultureCode.CSharp.Conductors.Tests.RepositoryConductorTests
+{
+
```
Comment (myty) [inline test/Conductors.Tests/RepositoryConductorTests/IRepositoryMock.cs:12]: nitpick (non-blocking): Extra line spacing

**Review by myty — APPROVED** [review]
Just a minor nitpick.  I'll approve.

File: test/Conductors.Tests/RepositoryConductorTests/IRepositoryMock.cs (hunk 1 above)
Comment (klauffer) [reply test/Conductors.Tests/RepositoryConductorTests/IRepositoryMock.cs:12]: will fix in next commit

**Pull request merged**